pkg/datapath/loader/compile.go

```diff
@@ -257,9 +257,23 @@ func compile(ctx context.Context, prog *progInfo, dir *directoryInfo, debug bool
 // * Assembly
 // * Object compiled with debug symbols
 func compileDatapath(ctx context.Context, dirs *directoryInfo, debug bool, logger *logrus.Entry) error {
-	// TODO: Consider logging kernel/clang versions here too
 	scopedLog := logger.WithField(logfields.Debug, debug)
 
+	versionCmd := exec.CommandContext(ctx, compiler, "--version")
+	compilerVersion, err := versionCmd.CombinedOutput(scopedLog, debug)
+	if err != nil {
+		return err
+	}
+	versionCmd = exec.CommandContext(ctx, linker, "--version")
+	linkerVersion, err := versionCmd.CombinedOutput(scopedLog, debug)
+	if err != nil {
+		return err
+	}
+	scopedLog.WithFields(logrus.Fields{
+		compiler: string(compilerVersion),
+		linker:   string(linkerVersion),
+	}).Debug("JoinEP: Tool versions")
```

Comment:
The CI looks for `JoinEP: ` as a failure string, which is why the CI failed. Suggestion below:

```suggestion
	}).Debug("Compiling datapath")
```